### ISSUE TITLE
NaN Gradients

### ISSUE THREAD

**Issue by g-luo** [facebookresearch/mmf#893]
## ❓ Questions and Help

Hello, 
When training my model, the loss steadily decreases then suddenly becomes NaN. I suspect that this is because it's hitting some malformed data sample somewhere (using my own custom news dataset). I was wondering if there are any known fixes for this (eg tossing a batch that results in NaN gradients) / gradient skipping implemented in MMF, especially considering I'm using gradient accumulation within MMF?

![Screen Shot 2021-04-20 at 11 43 30 PM](https://user-images.githubusercontent.com/36020488/115508520-41be7f00-a232-11eb-90ee-e1a9a948403f.png)

Thanks!

**Comment by g-luo** [comment]
Resolved! Thank you @ronghanghu for the solution. 
You can use https://github.com/facebookresearch/mmf/blob/master/mmf/modules/optimizers.py#L33-L34 and replace 

```
if p.grad.abs().sum().item() == 0:
     continue
```
with 
```
   if not torch.all(p.grad.isfinite()):
      continue
```

**Comment by apsdehal** [comment]
Hey,
In general, other way is to enable anomaly detection by using `training.detect_anomaly=True`. This will catch exact place where NaNs are happening. (Though it will slow the code down somewhat).

**Comment by g-luo** [comment]
Thanks for the pointer!
